src/Segments.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OneSignal;
+
+use OneSignal\Dto\Segment\CreateSegment;
+use OneSignal\Dto\Segment\ListSegments;
+
+class Segments extends AbstractApi
+{
+    public function __construct(OneSignal $client)
+    {
+        parent::__construct($client);
+    }
+
+    /**
+     * Get information about all segments.
+     *
+     * Application authentication key and ID must be set.
+     */
+    public function list(ListSegments $listSegmentsDto): array
+    {
+        $appId = $this->client->getConfig()->getApplicationId();
+
+        $request = $this->createRequest('GET', '/apps/'.$appId.'/segments?'.http_build_query($listSegmentsDto->toArray()));
+        $request = $request->withHeader('Authorization', "Basic {$this->client->getConfig()->getApplicationAuthKey()}");
+
+        return $this->client->sendRequest($request);
```

Comment:
By the way I think we should also introduce response models, so instead of plain arrays you get objects with proper data.
For example segments has `created_at`, `updated_at` so we should expose these as DateTimeImmutable from the model